### ISSUE TITLE
Mac support?

### ISSUE THREAD

**Issue by altitudems** [gravitl/netmaker#136]
Any chance we might see Mac/OSX support in the roadmap?

It doesn't appear that the client script will run.

Is it possible to add a manual peer perhaps?

My specific use case is wanting to join a network from my Macbook Pro.

Already using it for site-to-site and a gateway to lan. Working great.

Thanks so much for a great project!

**Comment by afeiszli** [comment]
Hi @altitudems can you please provide more info on your use case? This will help with our roadmap.

The issue is for full mesh support, we need to rewrite the client for each OS (which is why we just support systemd linux right now). True meshed multi-platform clients are probably a couple months away.

However, because of all the existing tools out there for "standard" wireguard VPN's, we could very easily implement a gateway config generator to allow devices such as Mac/iPhone/Windows to "hook into" the network, similar to the existing gateway mode.

This would not be a "full mesh" because other devices on the network would not be able to reach the Mac/iPhone or whatever, but the Mac/iPhone would be able to reach everything in the network. I believe this covers 90% of the use cases and we could implement this in a few weeks.

Would that fulfill your use case, or do you need other devices to be able to reach your Mac?

TL;DR: There's an easy way and a hard way, wondering which one you need.

**Comment by altitudems** [comment]
I think enabling existing wireguard clients would be perfect. In my case I don't need full mesh, but more of a "VPN in" kind of thing.

Essentially, I have a VPS, acting as a wireguard gateway to a private network. 

What I would like to be able to do is add new client peers to the network, possibly existing wireguard clients, so that for example a developer's device could access private network resources through the gateway. I don't have a need for other network peers to see the developer's device.

So almost exactly what you describe as the "standard" wire guard VPN.

Thanks so much for the reply and consideration!

**Comment by afeiszli** [comment]
Okay thank you, we will get that into the roadmap and hopefully have it as a feature by v0.5 (within a month).

**Comment by pete1019** [comment]
This has been added already with "External Clients", right?
Thank you!

**Comment by afeiszli** [comment]
Yes, you can support Mac via the Ingress Gateway. However, I'm leaving this open as a reminder that eventually we need to get the full client working with Mac.

**Comment by afeiszli** [comment]
added in 0.8
